pkg/apis/crd/v1alpha1/types.go

```diff
@@ -618,3 +627,8 @@ type ICMPProtocol struct {
 	ICMPType *int32 `json:"icmpType,omitempty"`
 	ICMPCode *int32 `json:"icmpCode,omitempty"`
 }
+
+type IGMPProtocol struct {
+	IGMPType     *int32 `json:"igmpType"`
```

Comment:
why not omitempty?

Reply:
fixed

Reply:
If it is not optional, just use `int32`

Reply:
it can be optional, and I just added `omitempty`